docs/quickstart.rst

```diff
@@ -61,198 +199,276 @@ instance had already existing in the destination directory, this command
-appears.
+.. code-block:: bash
 
-A client, whether a web server or just psql, can list multiple
-hosts in its PostgreSQL connection string, and use the parameter
-``target_session_attrs`` to add rules about which server to choose.
+   watch -d \
+     ssh -l ha-admin `vm_ip monitor` pg_autoctl show state --pgdata monitor
 
-To discover the url to use in our case, the following command can be used:
+In another terminal we’ll turn off the virtual server.
 
 .. code-block:: bash
 
-   pg_autoctl show uri --formation default --pgdata ./monitor
-   postgres://127.0.0.1:6010,127.0.0.1:6011/?target_session_attrs=read-write
+   az vm stop \
+     --resource-group ha-demo \
+     --name ha-demo-b
 
-Here we ask to connect to either node A or B -- whichever supports reads and
-writes:
+After a number of failed attempts to talk to node B, the monitor determines
+the node is unhealthy and puts it into the "demoted" state.  The monitor
+promotes node A to be the new primary.
 
 .. code-block:: bash
 
-   psql \
-     'postgres://127.0.0.1:6010,127.0.0.1:6011/?target_session_attrs=read-write'
+   .
+                              Name |   Port | Group |  Node |     Current State |    Assigned State
+   --------------------------------+--------+-------+-------+-------------------+------------------
+   ha-demo-a.internal.cloudapp.net |   5432 |     0 |     1 |      wait_primary |      wait_primary
+   ha-demo-b.internal.cloudapp.net |   5432 |     0 |     2 |           demoted |        catchingup
+
+Node A cannot be considered in full "primary" state since there is no secondary
+present, but it can still serve client requests. It is marked as "wait_primary"
+until a secondary appears, to indicate that it's running without a backup.
 
-When nodes A and B were both running, psql would connect to node A
-because B would be read-only. However now that A is offline and B is
-writeable, psql will connect to B. We can insert more data:
+Let's add some data while B is offline.
 
-.. code-block:: sql
+.. code-block:: bash
 
-   -- on the prompt from the psql command above:
-   insert into foo select generate_series(1000001, 2000000);
+   # notice how $APP_DB_URI continues to work no matter which node
+   # is serving as primary
+   ssh -l ha-admin -t `vm_ip app` -- \
+     psql "\"$APP_DB_URI\"" \
+       -c "\"INSERT INTO foo SELECT generate_series(1000001, 2000000);\""
 
-Resurrect node A
+Resurrect node B
 ----------------
 
-Let’s increase the disk space for node A, so it's able to run again.
+Run this command to bring node B back online:
 
 .. code-block:: bash
 
-   rm /mnt/node_a/bigfile
+   az vm start \
+     --resource-group ha-demo \
+     --name ha-demo-b
 
-Now the next time the keeper retries, it brings the node back. Node A
-goes through the state "catchingup" while it updates its data to match
-B. Once that's done, A becomes a secondary, and B is now a full primary.
+Now the next time the keeper retries its health check, it brings the node back.
```

Comment:
The health check is done only from the monitor. The other way around, we call it the “node active” protocol, the keeper asks the monitor “what now?” basically, and the monitor says “oh you've been demoted”.